### DISCUSSION TITLE
Can't deactivate Unity license

### DISCUSSION THREAD

**Discussion by BobsiUnity** [codemagic-ci-cd/codemagic-docs#2049] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Unity

### Steps to reproduce

I am solely attempting to build my Unity game for windows. 

### Expected results

For the build to complete.

### Actual results

It is failing on deactivating the license, stating it is not activated, however, I have activated it in Unity hub.

```
[Licensing::Module] Channel doesn't exist: "LicenseClient-builder"
[Licensing::Module] Successfully launched the LicensingClient (PId: 4688)
[Licensing::Module] Successfully connected to LicensingClient on channel: "LicenseClient-builder" (connect: 0.51s, validation: 0.06s, handshake: 0.66s)
[Licensing::Module] Connected to LicensingClient (PId: 4688, launch time: 0.00, total connection time: 1.23s)
Entitlement-based licensing initiated
[LicensingClient] Licenses updated successfully

LICENSE SYSTEM [2023915 20:24:29] Failed to return license: Unity license has not been activated, and cannot be returned.
Exiting without the bug reporter. Application will terminate with return code 1

Build failed :|
Step 5 script `Deactivate Unity License` exited with status code 1
```

### Build id (optional)

6504bd034ffa7e7b79a29233

**Comment by rafeysup93** [comment]
Hi @BobsiUnity, please note that you should contact our team through the chat widget in the UI for a quicker response, as your team is on a paid plan.

**Reply by rafeysup93** [reply]
You can also confirm on the Unity Hub if the activation script worked as expected and whether it activated the license in the first place.

**Reply by BobsiUnity** [reply]
How do I confirm on the Unity hub that the activation worked? I can see that my License is working in Unity Hub if that is what you mean.

**Reply by rafeysup93** [reply]
Once you go to this [link](https://id.unity.com/en/subscriptions), You can find **My Seats** under the **My Account** tab on the left side of the screen. Once you open that, you will see the activated seats below in the middle of your screen.

**Reply by BobsiUnity** [reply]
Yeah, I can see the seat as assigned.

**Reply by rafeysup93** [reply, marked as answer]
This conversation was continued on Intercom. The solution was to replace the License activation script with this working one.

```
- name: Activate & Build Unity Using a Command Prompt
      script: |
        $Serial = @($env:UNITY_SERIAL)
        $Email = @($env:UNITY_EMAIL)
        $Password = @($env:UNITY_PASSWORD)            
          Write-Host "Attempting activation for a license with the following details:"
          Write-Host "Serial: $Serial"
          Write-Host "Email: $Email"
          Write-Host "Password: $Password"
          $process = Start-Process -FilePath "$env:UNITY_HOME\Unity.exe" -Wait -PassThru -ArgumentList "-batchmode -nographics -quit 
            -logFile -username $currentEmail -password $currentPassword -serial $currentSerial"
            if ($process.ExitCode -eq 0) {
                Write-Host "Unity activated successfully"
                exit 0
            } else {
                Write-Host "Activation failed"
            }
        
        Write-Host "All license activations failed!"
        exit 1
```
